astropy/io/fits/hdu/base.py

```diff
@@ -1311,55 +1285,39 @@ def verify_checksum(self, blocking='standard'):
 
         if 'CHECKSUM' in self._header:
             if 'DATASUM' in self._header:
-                datasum = self._calculate_datasum(blocking)
+                datasum = self._calculate_datasum()
             else:
                 datasum = 0
-            checksum = self._calculate_checksum(datasum, blocking)
+            checksum = self._calculate_checksum(datasum)
             if checksum == self._header['CHECKSUM']:
                 return 1
-            elif blocking == 'either':
-                # i.e. standard failed,  try nonstandard
-                return self.verify_checksum(blocking='nonstandard')
             else:
-                # Failed with all permitted blocking kinds
+                # Failed
                 return 0
         else:
             return 2
 
-    def _verify_checksum_datasum(self, blocking):
+    def _verify_checksum_datasum(self):
         """
         Verify the checksum/datasum values if the cards exist in the header.
         Simply displays warnings if either the checksum or datasum don't match.
         """
 
-        # NOTE:  private data members _checksum and _datasum are
-        # used by the utility script "fitscheck" to detect missing
-        # checksums.
-
         if 'CHECKSUM' in self._header:
             self._checksum = self._header['CHECKSUM']
-            self._checksum_comment = self._header.comments['CHECKSUM']
-            if not self.verify_checksum(blocking):
+            self._checksum_valid = self.verify_checksum()
+            if not self._checksum_valid:
                 warnings.warn(
                     'Checksum verification failed for HDU {0}.\n'.format(
                         (self.name, self.ver)), AstropyUserWarning)
-        else:
-            self._checksum = None
-            self._checksum_comment = None
 
         if 'DATASUM' in self._header:
             self._datasum = self._header['DATASUM']
-            self._datasum_comment = self._header.comments['DATASUM']
-
-            if not self.verify_datasum(blocking):
+            self._datasum_valid = self.verify_datasum()
```

Comment:
The datasum gets computed twice since it is also computed within `verify_checksum`. It would be nice if there was a way to store the datasum value between calls so that it does not have to be recomputed.